core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala

```diff
@@ -36,272 +48,242 @@ object ReactTable {
-    .render { $ =>
-      val (row, props) = $.props
-      <.div(
-        props.style.tableRow,
-        if (props.config.nonEmpty) {
-          props.columns.map { item =>
-            val cell = getColumnDiv(item, props.config)
-            val f    = getRenderFunction(item, props.config)
-            if (f.isDefined) cell(f.get(row(item)))
-            else cell(row(item).toString)
-          }
-        }.toTagMod
-        else
-          props.columns.map { item =>
-            <.div(row(item).toString)
-          }.toTagMod
-      )
-    }
-    .build
+    TagMod(^.padding := "1px")
 
   val tableC = ScalaComponent
     .builder[(Props, State, Backend)]("table")
-    .render { $ =>
+    .render { $ ⇒
       val (props, state, b) = $.props
-      val rows = state.filteredModels
+
+      def renderHeader: TagMod =
+        <.tr(
+          props.style.tableHeader,
+          props.configs.zipWithIndex.map {
+            case (config, columnIndex) ⇒
+              val cell = getHeaderDiv(config)
+              config.sortBy.fold(cell(config.name.capitalize))(sortByFn =>
+                cell(
+                  ^.cursor := "pointer",
+                  ^.onClick --> b.sort(sortByFn, columnIndex),
+                  config.name.capitalize,
+                  props.style
+                    .sortIcon(state.sortedState.isDefinedAt(columnIndex) && state.sortedState(
+                      columnIndex) == asc)
+                    .when(state.sortedState.isDefinedAt(columnIndex))
+              ))
+          }.toTagMod
+        )
+
+      def renderRow(model: T): TagMod =
+        <.tr(
+          props.style.tableRow,
+          props.configs
+            .map(
+              config =>
+                <.td(^.whiteSpace.nowrap.when(config.nowrap),
+                     ^.verticalAlign.middle,
+                     config.cellRenderer(model)))
+            .toTagMod
+        )
+
+      val rows = state.filteredData
         .slice(state.offset, state.offset + state.rowsPerPage)
         .zipWithIndex
         .map {
-          case (row, i) => tableRow.withKey(i)((row, props))
+          case (row, i) ⇒ renderRow(row) //tableRow.withKey(i)((row, props))
         }
-      <.div(props.style.table)(
-        tableHeader((props, b, state)),
-        rows.toTagMod
-      )
+        .toTagMod
+
+      <.div(props.style.table, <.table(<.thead(renderHeader()), <.tbody(rows)))
     }
     .build
 
-  val settingsBar = ScalaComponent
-    .builder[(Props, Backend, State)]("settingbar")
-    .render { $ =>
-      val (p, b, s)             = $.props
-      var value                 = ""
-      var options: List[String] = Nil
-      val total                 = s.filteredModels.length
-      if (total > p.rowsPerPage) {
-        value = s.rowsPerPage.toString
-        options = immutable.Range
-          .inclusive(p.rowsPerPage, total, 10 * (total / 100 + 1))
-          .:+(total)
-          .toList
-          .map(_.toString)
+  val settingsBar =
```

Comment:
So there is one decision we should make for generic components like this. This has a somewhat big problem with performance: Since it's wrapped in a case class these scala components will be "new" components (from react's view) everytime that class is reinstantiated. This is both surprising and difficult to debug.

Our options are (afaik):
 - do nothing, keep it like it is.
- Rewrite `ReactTable` to an *abstract* class, and encourage users to create object where they specify types and capture typeclass instances. Provide a default `object` for a normal case, say some stringly based variant. Create a new react component instance for each type (cheap enough - at least it will re-render well)
- Create the react component in a static context, say an `object` private to the component, and, ahem, cast it to the correct type when the user instantiates of `case class ReactTable`.

Any thoughts? I'm willing to move this to a separate issue if you just want to be done with this PR :)